pkg/controller/v1alpha1/inferencegraph/raw_ig.go

```diff
@@ -0,0 +1,150 @@
+package inferencegraph
+
+import (
+	"encoding/json"
+	"github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+	"github.com/kserve/kserve/pkg/controller/v1beta1/inferenceservice/reconcilers/raw"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"strconv"
+
+	v1alpha1api "github.com/kserve/kserve/pkg/apis/serving/v1alpha1"
+	"github.com/kserve/kserve/pkg/constants"
+	v1 "k8s.io/api/core/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"strings"
+)
+
+var logger = logf.Log.WithName("InferenceGraphRAWDeploymentMode")
+
+func createInferenceGraphPodSpec(graph *v1alpha1api.InferenceGraph, config *RouterConfig) *v1.PodSpec {
+	bytes, err := json.Marshal(graph.Spec)
+	if err != nil {
+		return nil
+	}
+
+	podSpec := &v1.PodSpec{
+		Containers: []v1.Container{
+			{
+				Name:  graph.ObjectMeta.Name,
+				Image: config.Image,
+				Args: []string{
+					"--graph-json",
+					string(bytes),
+				},
+				Resources: constructResourceRequirements(*graph, *config),
+			},
+		},
+		Affinity: graph.Spec.Affinity,
+	}
+
+	// Only adding this env variable "PROPAGATE_HEADERS" if router's headers config has the key "propagate"
+	value, exists := config.Headers["propagate"]
+	if exists {
+		podSpec.Containers[0].Env = []v1.EnvVar{
+			{
+				Name:  constants.RouterHeadersPropagateEnvVar,
+				Value: strings.Join(value, ","),
+			},
+		}
+	}
+
+	return podSpec
+}
+
+func constructGraphObjectMeta(name string, namespace string) metav1.ObjectMeta {
+	objectMeta := metav1.ObjectMeta{
+		Name:        name,
+		Namespace:   namespace,
+		Labels:      make(map[string]string),
+		Annotations: make(map[string]string),
+	}
+
+	return objectMeta
+}
+
+func constructGraphComponentExtensionSpec(annotations map[string]string) v1beta1.ComponentExtensionSpec {
```

Comment:
I feel we should not use this `ComponentExtensionSpec` from v1beta package as this is intended for ISVCs components (predictor/transformer) 
Can we create custom IGExtensionSpec struct, under v1alpha1 package, to keep ISVCs spec separate from IG spec ?

Reply:
@rachitchauhan43 This is a great suggestion.  But I see a challenge here.  
ComponentExtensionSpec exists all the way from spec yaml for Preditor, Transformer etc. 

However if I introduce IGExtensionSpec, this is not really going into Inference graph spec.  Also I cannot reuse code in v1beta1.inferenceservice.reconcilers as is - good amount of refactoring is required. 

Currently I am simply using v1beta1.ComponentExtensionSpec to simply pass through the annotations mentioned in Inference graph.   
```
    serving.kserve.io/class: "hpa"
    serving.kserve.io/max-scale: "7"
    serving.kserve.io/metric: rps
    serving.kserve.io/min-scale: "1"
    serving.kserve.io/target: "40"
```

What I can do instead is, to use the ObjectMeta itself to propagate the HPA related annotations and change v1beta1.reconcilers to construct AutoScaler object using values from Annotations rather than ComponentExtensionSpec